### PR TITLE
xserver service: add xinitrc window manager

### PR DESCRIPTION
**Pull request by robgssp**
Add a "window manager" to run the user's .xinitrc on login

### CODE REVIEW

**Comment by jagajaga** [comment]
You can use `~/.xsession` instead of `~/.xinitrc` AFAIK.

File: nixos/modules/services/x11/window-managers/xinitrc.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+with lib;
+
+let
+  cfg = config.services.xserver.windowManager.xinitrc;
+in
+
+{
+  options = {
+    services.xserver.windowManager.xinitrc = {
+      enable = mkOption {
```
Comment (nbp) [inline nixos/modules/services/x11/window-managers/xinitrc.nix]: Use `mkEnableOption` instead, this is just a short-cut over the boiler plate of `enable = mkOption { …`.

**Comment by nbp** [comment]
This change looks good to me.
Also, based on what I can read [1], it seems that `~/.xinitrc` is used when the user start an `xserver` him self, while the `~/.xsession` is when the user login with a display manager.

[1] http://unix.stackexchange.com/a/47426

**Comment by robgssp** [comment]
Actually it appears @jagajaga's correct: without this change, just disable the default `xterm` desktop manager and `~/.xsession` will get run.

This should probably be documented somewhere.

**Comment by robgssp** [comment]
Correction, if `~/.xsession` exists the system-configured window/session managers are ignored. The behavior's fine, but it should be noted in the manual.

**Comment by magnetophon** [comment]
I created an `~/.xsession`, containing
``
#!/usr/bin/env bash
exec urxvtc
``
but I'm going to my normal i3 environment.

Am I doing it wrong? Is this not available in 15.09?

**Comment by jagajaga** [comment]
Didn't you forget to `chmod +x` it?

**Comment by magnetophon** [comment]
@jagajaga
Yup, rooky mistake!  Thanks!

**Comment by magnetophon** [comment]
When I use this method, my `shellInit` is not called.
I'm forced to uses nano, eeeeeeew!

Is that expected behaviour, and if so, why, and how do I work around it?

**Comment by robgssp** [comment]
Source your .profile from .xsession.

**Comment by magnetophon** [comment]
@robgssp 
I dont have a `~/.profile`.
Should it have been created by nixos?
I prefer to set `$EDITOR` and such only once, in configuration.nix

**Comment by robgssp** [comment]
@magnetophon: try using `#!/bin/sh -l` as your shebang line. That should load whatever usually gets loaded at login.

**Comment by magnetophon** [comment]
@robgssp
thanks!

**Comment by mogorman** [comment]
Should this just be deleted and using .xsession?

**Comment by robgssp** [comment]
@mogorman: yes.

**Pull request closed without merging**